src/app/geo/geochart.map.component.ts

```diff
@@ -49,19 +52,30 @@ export class GeochartMapComponent implements OnInit, OnChanges, AfterViewInit {
       distinctUntilChanged()).subscribe(() => this.drawMap());
   }
 
+  getMapWidthOrHeight(): {width?: number, height?: number} {
+    if (this.windowSize && this.windowSize.width && this.windowSize.width <= 768) {
+      return {width: this.windowSize.width - 40};
+    } else {
+      return {height: 320};
+    }
+  }
+
+  toGoogleDataTable(ranks: Rank[]) {
+    const data = new google.visualization.DataTable();
+    data.addColumn('string', getGroupName(this.routerParams.metricsType, this.routerParams.group));
+    data.addColumn('number', 'Downloads');
+    data.addRows(ranks.map(d => [{v: d.code, f: d.label}, d.total]));
+    return data;
+  }
+
   drawMap() {
-    if (GeochartMapComponent.googleLoaded && google.visualization && this.data) {
-      let width, height;
-      if (this.windowSize && this.windowSize.width && this.windowSize.width <= 768) {
-        width = this.windowSize.width - 40;
-      } else {
-        height = 320;
-      }
+    if (GeochartMapComponent.googleLoaded && google.visualization) {
       const options = {
         displayMode: 'region',
         colorAxis: {colors: this.colors},
         legend: 'none'
       };
+      const { width, height } = this.getMapWidthOrHeight();
```

Comment:
This logic and the following two `if` statements could be combined:
`{width: options['width'], height: options['height']} = this.getMapWidthOrHeight();`

Reply:
... or `Object.assign(options, this.getMapWidthOrHeight())`

Debatable whether either of those is an improvement.

Reply:
good call 👍